File: Games/NoughtsAndCrosses/src/main/java/jokatu/game/games/noughtsandcrosses/player/NoughtsAndCrossesPlayer.java
package jokatu.game.games.noughtsandcrosses.player;

import jokatu.game.games.noughtsandcrosses.input.NoughtOrCross;
import jokatu.game.player.AbstractPlayer;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class NoughtsAndCrossesPlayer extends AbstractPlayer {

	private NoughtOrCross allegiance;

	NoughtsAndCrossesPlayer(String username) {
		super(username);
	}

	@Nullable
	public NoughtOrCross getAllegiance() {
		return allegiance;
	}

	public void setAllegiance(@NotNull NoughtOrCross noughtOrCross) {
		allegiance = noughtOrCross;
	}
}

File: Games/NoughtsAndCrosses/src/main/kotlin/jokatu/game/games/noughtsandcrosses/player/NoughtsAndCrossesPlayer.kt
package jokatu.game.games.noughtsandcrosses.player

import jokatu.game.games.noughtsandcrosses.input.NoughtOrCross
import jokatu.game.player.AbstractPlayer

class NoughtsAndCrossesPlayer(username: String) : AbstractPlayer(username) {
	var allegiance: NoughtOrCross? = null
}
